### PR TITLE
Prevent Input w/ Copy button from submitting forms

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

This fixes a bug where clicking the copy button on the InputWithCopy component would submit a form. The main reason was due to the `<button>` not having a `type` explicitly set, which would result in a form submitting if it was inside of one. To fix that I've:

* Explicitly set `button.type=button`
* Added a `preventDefault()` on the click handler as an extra measure to prevent side-effects beyond copying to the clipboard.

I've also updated a few things in the InputWithCopy component to re-use existing hooks/components. A byproduct of that was updating a couple related components.

### New `<Button>` features:
* New optional `spacing='compact|default'` prop to allow tighter padding when needed.
* Added `type='transparent'` instead of the `button.reset` global css class name

### `<Modal>` changes
* Added optional `disableAutoFocus=true|false` prop to opt-out of the modal auto-focusing the first focusable element. Useful in this particular use-case.

### Normalized `<TextInputField>`, `<SelectInputField>` & `<CheckboxInputField>`
* Each have `topMargin` and `containerClassName` for consistency

## Related Issue(s)

WEB-581

## How to test

* Open the Git Providers Settings for your org.
* Create a new provider.
* Click the Copy icon on the Redirect URL and ensure it doesn't submit the modal/form.

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - brad-web-55cceb5ef3e</li>
	<li><b>🔗 URL</b> - <a href="https://brad-web-55cceb5ef3e.preview.gitpod-dev.com/workspaces" target="_blank">brad-web-55cceb5ef3e.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - brad-web-581-copying-gitprovider-redirect-url-triggers-activate-action-gha.12579</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/components/Tooltip.tsx [hunk 1]
```diff
@@ -47,7 +48,14 @@ function Tooltip(props: TooltipProps) {
     }, [showTooltipTimeout]);
 
     return (
-        <span onMouseLeave={handleMouseLeave} onMouseEnter={handleMouseEnter} ref={setTriggerEl}>
+        <span
+            onMouseEnter={handlShow}
+            onFocus={handlShow}
```
Comment (selfcontained) [inline components/dashboard/src/components/Tooltip.tsx:53]: This allows tooltips to show when tabbing into the component.

File: components/dashboard/src/user-settings/Account.tsx [hunk 2]
```diff
@@ -159,12 +160,9 @@ function ProfileInformation(props: {
                         }}
                     />
                     {props.user && (
-                        <div className="flex flex-col space-y-2 mt-4">
-                            <label className={"text-md font-semibold dark:text-gray-400 text-gray-600"}>User ID</label>
-                            <p className={"text-sm text-gray-500 dark:text-gray-500"}>
-                                <InputWithCopy className="max-w-md w-32" value={props.user.id} tip="Copy User ID" />
-                            </p>
-                        </div>
+                        <InputField label="User ID">
```
Comment (selfcontained) [inline components/dashboard/src/user-settings/Account.tsx:163]: Swapping to use components for consistency

**Comment by svenefftinge** [comment]
The dialog immediately shows like this without any hover or so.
Screenshot 2023-07-04 at 08 33 30

**Comment by svenefftinge** [comment]
Btw what I really like and find satisfying is the copy hash button on github, which turns into a checkbox after I clicked it.

**Review by svenefftinge — APPROVED** [review]

**Pull request merged**